agent/create_delete_topics_test.go

```diff
@@ -0,0 +1,241 @@
+	require.NoError(t, err)
+	defer func() {
+		err := conn.Close()
+		require.NoError(t, err)
+	}()
+	topicName := "test-topic-1"
+	//Create
+	req := kafkaprotocol.CreateTopicsRequest{
+		Topics: []kafkaprotocol.CreateTopicsRequestCreatableTopic{
+			{
+				Name:              &topicName,
+				NumPartitions:     23,
+			},
+		},
+	}
+	var resp kafkaprotocol.CreateTopicsResponse
+	r, err := conn.SendRequest(&req, kafkaprotocol.APIKeyCreateTopics, 5, &resp)
+	require.NoError(t, err)
+	createResp, ok := r.(*kafkaprotocol.CreateTopicsResponse)
+	require.True(t, ok)
+	require.Equal(t, 1, len(createResp.Topics))
+	require.Equal(t, topicName, common.SafeDerefStringPtr(createResp.Topics[0].Name))
+	require.Equal(t, int16(0), createResp.Topics[0].ErrorCode)
+	require.Equal(t, int32(23), createResp.Topics[0].NumPartitions)
+	info, topicExists, err := agent.topicMetaCache.GetTopicInfo(topicName)
+	require.True(t, topicExists)
+	require.NoError(t, err)
+	require.Equal(t, topicName, info.Name)
+	require.Equal(t, 23, info.PartitionCount)
+	require.Equal(t, 7*24*time.Hour, info.RetentionTime)
+	//Duplicate create
+	req = kafkaprotocol.CreateTopicsRequest{
+		Topics: []kafkaprotocol.CreateTopicsRequestCreatableTopic{
+			{
+				Name:              &topicName,
+				NumPartitions:     23,
+			},
+		},
+	}
+	var resp2 kafkaprotocol.CreateTopicsResponse
+	r2, err := conn.SendRequest(&req, kafkaprotocol.APIKeyCreateTopics, 5, &resp2)
+	require.NoError(t, err)
+	createResp2, ok := r2.(*kafkaprotocol.CreateTopicsResponse)
+	require.True(t, ok)
+	require.Equal(t, 1, len(createResp2.Topics))
+	require.Equal(t, topicName, common.SafeDerefStringPtr(createResp2.Topics[0].Name))
+	require.Equal(t, int16(kafkaprotocol.ErrorCodeTopicAlreadyExists), createResp2.Topics[0].ErrorCode)
+	require.Equal(t, int32(23), createResp2.Topics[0].NumPartitions)
+}
+
+func TestDeleteNonExistentTopic(t *testing.T) {
+	cfg := NewConf()
+	agent, _, tearDown := setupAgentWithoutTopics(t, cfg)
+	defer tearDown(t)
+	cl, err := NewKafkaApiClient()
+	require.NoError(t, err)
+	conn, err := cl.NewConnection(agent.Conf().KafkaListenerConfig.Address)
+	require.NoError(t, err)
+	defer func() {
+		err := conn.Close()
+		require.NoError(t, err)
+	}()
+	topicName := "test-topic-1"
+	req := kafkaprotocol.DeleteTopicsRequest{
+		TopicNames: []*string{
+			&topicName,
+		},
+	}
+	var resp kafkaprotocol.DeleteTopicsResponse
+	r, err := conn.SendRequest(&req, kafkaprotocol.APIKeyDeleteTopics, 5, &resp)
+	require.NoError(t, err)
+	deleteResp, ok := r.(*kafkaprotocol.DeleteTopicsResponse)
+	require.True(t, ok)
+	require.Equal(t, 1, len(deleteResp.Responses))
+	require.Equal(t, topicName, *deleteResp.Responses[0].Name)
+	require.Equal(t, int16(kafkaprotocol.ErrorCodeUnknownTopicOrPartition), deleteResp.Responses[0].ErrorCode)
+	_, topicExists2, _ := agent.topicMetaCache.GetTopicInfo(topicName)
+	require.False(t, topicExists2)
+}
+
+func TestInvalidTopicName(t *testing.T) {
```

Comment:
This test can be parameterised with the topic name, and then it can be tested with various different topic names, not just one.

Reply:
makes sense. Done.